modin/pandas/general.py

```diff
@@ -404,26 +404,34 @@ def concat(
                 sort=sort,
             )
         )
-    if join not in ["inner", "outer"]:
+    if join == "outer":
+        # Filter out empties
+        list_of_objs = [
+            obj
+            for obj in list_of_objs
+            if (
+                isinstance(obj, (Series, pandas.Series))
```

Comment:
Why is this part of the condition not checked for `lazy_execution` for Modin Series?

Reply:
We should not filter out Series - https://github.com/pandas-dev/pandas/blob/main/pandas/core/reshape/concat.py#L476

Reply:
For Series we have a different code branch https://github.com/modin-project/modin/blob/b113f47a1787a68b3846d861dd2526e38b1bdc72/modin/pandas/general.py#L391 where we don't filter them.

Reply:
This branch checks if all objs are modin Series, but here we filter out empty DataFrames, while preserving empty Series.

Reply:
> We should not filter out Series - https://github.com/pandas-dev/pandas/blob/main/pandas/core/reshape/concat.py#L476

Judging by your link, empty Series are not filtered only when all objects are Series types. Isn't it?

Reply:
According to the following condition, any Series are not filtered out:

```python
            # filter out the empties if we have not multi-index possibilities
            # note to keep empty Series as it affect to result columns / name
            non_empties = [
                obj for obj in objs if sum(obj.shape) > 0 or isinstance(obj, ABCSeries)
            ]
```

Reply:
```python
        # figure out what our result ndim is going to be
        ndims = set()
        for obj in objs:
            if not isinstance(obj, (ABCSeries, ABCDataFrame)):
                msg = (
                    f"cannot concatenate object of type '{type(obj)}'; "
                    "only Series and DataFrame objs are valid"
                )
                raise TypeError(msg)

            ndims.add(obj.ndim)

        # get the sample
        # want the highest ndim that we have, and must be non-empty
        # unless all objs are empty
        sample: NDFrame | None = None
        if len(ndims) > 1:
            max_ndim = max(ndims)
            for obj in objs:
                if obj.ndim == max_ndim and np.sum(obj.shape):
                    sample = obj
                    break

        else:
            # filter out the empties if we have not multi-index possibilities
            # note to keep empty Series as it affect to result columns / name
            non_empties = [
```
We get into `else` branch only if the input contains concatenation objects of the same types. These are either all dataframes, but this has nothing to do with series, or all series and this is exactly the case you are talking about. We also have a separate branch for series, in which objects are not filtered in the same way as in pandas. I want to say that from this point of view, the already existing code should correspond to it. Am I missed something?